rust/agama-lib/share/profile.schema.json

```
@@ -224,54 +219,314 @@
-          "description": "Root password",
+          "title": "Root password",
           "type": "string"
         },
         "sshPublicKey": {
-          "description": "SSH public key",
+          "title": "SSH public key",
           "type": "string"
         }
       }
     },
     "localization": {
-      "description": "Localization settings",
+      "title": "Localization settings",
       "type": "object",
       "properties": {
         "language": {
-          "description": "System language ID (e.g., 'en_US')",
+          "title": "System language ID (e.g., 'en_US')",
           "type": "string"
         },
         "keyboard": {
-          "description": "Keyboard layout ID",
+          "title": "Keyboard layout ID",
           "type": "string"
         },
         "timezone": {
-          "description": "Time zone identifier such as 'Europe/Berlin'",
+          "title": "Time zone identifier such as 'Europe/Berlin'",
           "type": "string"
         }
       }
     },
     "storage": {
-      "description": "Storage settings",
+      "title": "Storage settings",
       "type": "object",
+      "additionalProperties": false,
       "properties": {
-        "bootDevice": {
-          "description": "Device used for booting (e.g., '/dev/sda'). By default, all file systems are created in the boot device.",
-          "type": "string"
-        },
-        "lvm": {
-          "description": "Whether LVM is used.",
-          "type": "boolean"
-        },
-        "encryptionPassword": {
-          "description": "If set, the devices are encrypted using the given password.",
-          "type": "string"
+        "guided": {
+          "title": "Settings to execute a Guided Proposal",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "target": {
+              "title": "Target device",
+              "oneOf": [
+                {
+                  "enum": ["disk", "newLvmVg"]
+                },
+                {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": { "disk": { "type": "string" } },
+                  "required": ["disk"]
+                },
+                {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": {
+                    "newLvmVg": {
+                      "type": "array",
+                      "items": { "type": "string" }
+                    }
+                  },
+                  "required": ["newLvmVg"]
+                }
+              ]
+            },
+            "boot": {
+              "title": "Configuration of the boot settings",
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "configure": {
+                  "title": "Whether to configure partitions for booting",
+                  "type": "boolean"
```

```suggestion
                  "type": "boolean",
                  "default": "true"
```

https://json-schema.org/understanding-json-schema/reference/annotations